### PR TITLE
Consolidate clippy jobs

### PR DESCRIPTION
**Pull request by npmccallum**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -49,30 +49,6 @@ jobs:
...
-        with:
-          components: clippy
-          toolchain: nightly
-          profile: minimal
-          target: x86_64-unknown-linux-musl
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: --target x86_64-unknown-linux-musl --manifest-path=${{ matrix.crate.path }} -- -D warnings
```
Comment (haraldh) [inline .github/workflows/lint.yml:71]: `--target x86_64-unknown-linux-musl` is important

Comment (haraldh) [reply .github/workflows/lint.yml:71]: Which can't be done for the main `enarx-keep` crate... had some openssl issues

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -43,40 +43,18 @@ jobs:
...
-    strategy:
-      fail-fast: false
-      matrix:
-        crate:
           - {name: rcrt1, path: internal/rcrt1/Cargo.toml}
           - {name: sallyport, path: internal/sallyport/Cargo.toml}
           - {name: shim-sgx, path: internal/shim-sgx/Cargo.toml}
```
Comment (haraldh) [inline .github/workflows/lint.yml:54]: rcrt1 and shim-sgx also compile for this target ... please include

**Review by haraldh — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 2 above)
Comment (npmccallum) [reply .github/workflows/lint.yml]: fixed

**Review by haraldh — APPROVED** [review]

**Review by lkatalin — APPROVED** [review]

**Pull request merged**